ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v16/tasks/main.yml

```diff
@@ -13,37 +13,54 @@
   when: xlc16.stat.islnk is defined
   tags: xlc16
 
-- name: Transfer and Extract XLC16
-  unarchive:
-    src: /Vendor_Files/aix/XLC/XL_C_Cpp_FOR_AIX_V16.1_EMG.tar.Z
-    dest: /tmp
-    remote_src: no
-  when: xlc16.stat.islnk is not defined
-  tags: xlc16
+- name: Vendor File processing
+  block:
+    - name: Check for Vendor file
+      stat:
+        path: /Vendor_Files/aix/XLC/XL_C_Cpp_FOR_AIX_V16.1_EMG.tar.Z
+      register: _vendor
+    - name: Transfer and Extract XLC16
+      unarchive:
+        src: /Vendor_Files/aix/XLC/XL_C_Cpp_FOR_AIX_V16.1_EMG.tar.Z
+        dest: /tmp
+        remote_src: no
+      register: _images
+      when: _vendor.stat.exists == true
+      tags: xlc16
 
-- name: Install IBM XLC16 - installp
-  command: installp -aXYg -e /tmp/usr/install.log -d /tmp/usr/sys/inst.images all
-  register: result.xlc
-  ignore_errors: yes
-  when: xlc16.stat.islnk is not defined
-  tags: xlc16
+    - name: Install IBM XLC16 - installp
+      command: installp -aXYg -e /tmp/usr/install.log -d /tmp/usr/sys/inst.images all
+      register: _installp
+      ignore_errors: yes
+      when: _images.skipped == false
```

Comment:
I think the previous task has trouble registering "_images", despite the task succeeding, as it causes an error when trying to access "_images.skipped":
```
fatal: [p9-aix1-ojdk06.osuosl.org]: FAILED! => {"msg": "The conditional check '_images.skipped == false' failed. The error was: error while evaluating conditional (_images.skipped == false): 'dict object' has no attribute 'skipped'\n\nThe error appears to be in '/Users/hkhel/AdoptOpenJDK/openjdk-infrastructure/ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v16/tasks/main.yml': line 33, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Install IBM XLC16 - installp\n      ^ here\n"
```

Reply:
I think because the previous task `name: Transfer and Extract XLC16` wasnt skipped, ansible doesnt add the `.skipped` attribute to `_images`

Reply:
I'll have to fake something - for when the files are there. Will take some time.

Reply:
When not skipped, `variable`.skipped is not defined - so added a check that the variable is defined